numpy/f2py/crackfortran.py

```diff
@@ -2403,7 +2403,7 @@ def _selected_real_kind_func(p, r=0, radix=0):
     if p < 16:
         return 8
     machine = platform.machine().lower()
-    if machine.startswith('power') or machine.startswith('ppc64'):
+    if machine.startswith(('aarch64', 'power', 'ppc64', 's390x')):
```

Comment:
Also, not sure if this should be a strict check for the new ones instead of `startswith`.

Reply:
Probably more compiler dependent, I'd guess. Let's put this in and fix it up if problems ensue.